File: frontend/src/types/recommendation.ts
import type { Reel } from './reel'

export interface Explanation {
  similarity_score: number
  reasons: string[]
}

export interface RecommendedReel {
  reel: Reel
  score: number
  explanation: Explanation
}

export interface RecommendationsResponse {
  items: RecommendedReel[]
}

File: backend/app/schemas/recommendation.py
from pydantic import BaseModel

from app.schemas.reel import ReelOut


class ExplanationOut(BaseModel):
    similarity_score: float
    reasons: list[str]


class RecommendedReelOut(BaseModel):
    reel: ReelOut
    score: float
    explanation: ExplanationOut


class RecommendationsResponse(BaseModel):
    items: list[RecommendedReelOut]
